modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/classloader/ArtifactClassLoaderConfigurationBuilder.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.artifact.activation.internal.classloader;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptySet;
+import static java.util.stream.Collectors.toList;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.mule.runtime.module.artifact.api.descriptor.BundleDependency;
+import org.mule.runtime.module.artifact.api.descriptor.BundleDescriptor;
+import org.mule.runtime.module.artifact.api.descriptor.BundleScope;
+import org.mule.runtime.module.artifact.api.descriptor.ClassLoaderModel;
+import org.mule.tools.api.classloader.model.AppClassLoaderModel;
+import org.mule.tools.api.classloader.model.Artifact;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+
+/**
+ * ClassLoaderModelBuilder that adds the concept of Shared Library for the configured dependencies.
+ */
+public class ArtifactClassLoaderConfigurationBuilder extends ClassLoaderModel.ClassLoaderModelBuilder {
```

Comment:
would rename this to `ApplicationClassLoaderConfigurationBuilder`

Reply:
Renamed to `DeployableClassLoaderConfigurationBuilder`.